src/tools/rbd/action/Group.cc

```diff
@@ -756,6 +758,111 @@ int execute_group_snap_list(const po::variables_map &vm,
   return 0;
 }
 
+int execute_group_snap_info(const po::variables_map &vm,
+                            const std::vector<std::string> &ceph_global_args) {
+  size_t arg_index = 0;
+  std::string pool_name;
+  std::string namespace_name;
+  std::string group_name;
+  std::string group_snap_name;
+
+  int r = utils::get_pool_generic_snapshot_names(
+    vm, at::ARGUMENT_MODIFIER_NONE, &arg_index, at::POOL_NAME, &pool_name,
+    &namespace_name, GROUP_NAME, "group", &group_name, &group_snap_name, true,
+    utils::SNAPSHOT_PRESENCE_REQUIRED, utils::SPEC_VALIDATION_FULL);
+  if (r < 0) {
+    return r;
+  }
+
+  at::Format::Formatter formatter;
+  r = utils::get_formatter(vm, &formatter);
+  if (r < 0) {
+    return r;
+  }
+  Formatter *f = formatter.get();
+
+  librados::Rados rados;
+  librados::IoCtx io_ctx;
+  r = utils::init(pool_name, namespace_name, &rados, &io_ctx);
+  if (r < 0) {
+    return r;
+  }
+
+  librbd::RBD rbd;
+  librbd::group_snap_info2_t group_snap;
+
+  r = rbd.group_snap_get_info(io_ctx, group_name.c_str(),
+                              group_snap_name.c_str(), &group_snap);
+
+  if (r < 0) {
+    std::cerr << "rbd: failed to show group snapshot: "
+              << cpp_strerror(r) << std::endl;
+    return r;
+  }
+
+   int state = group_snap.state;
+   std::string state_string;
+   if (state == RBD_GROUP_SNAP_STATE_COMPLETE) {
+     state_string = "complete";
+   } else {
+     state_string = "incomplete";
+   }
+
+  if (f) {
+    f->open_object_section("group_snapshot");
+    f->dump_string("id", group_snap.id);
+    f->dump_string("name", group_snap.name);
+    f->dump_string("state", state_string);
+    f->dump_string("image_snap_name", group_snap.image_snap_name);
+  } else {
+    std::cout << "rbd group snapshot '" << group_snap.name << "':\n"
+              << "\tid: " << group_snap.id << std::endl
+              << "\tstate: " << state_string << std::endl;
+    if(!group_snap.image_snap_name.empty()){
+      std::cout << "\timage snap: " << group_snap.image_snap_name
+                << std::endl;
+    }
+  }
+
+  if (f) {
+    f->open_array_section("images");
+  } else {
+    std::cout << "\timages:" << std::endl;
+  }
+
+  for (const auto &i : group_snap.image_snaps) {
+    r = rados.pool_reverse_lookup(i.pool_id, &pool_name);
+    if (r < 0) {
+      std::cerr << "error looking up pool name for pool_id=" << i.pool_id
+                << ": " << cpp_strerror(r) << std::endl;
+    }
+
+    if (f) {
+      f->open_object_section("image");
+      f->dump_string("pool_name", pool_name);
```

Comment:
This will attempt to print pool_name even if the pool_reverse_lookup() call fails.